test/hotspot/jtreg/vmTestbase/nsk/share/jpda/DebugeeProcess.java

```diff
@@ -207,6 +207,12 @@ public int waitFor () {
         if (process != null) {
             process.destroy();
         }
+        if (pipe != null) {
+            pipe.close();
+        }
+        if (binder != null) {
+            binder.close();
+        }
```

Comment:
Won't this be skipped if there is an exception during `waitForDebugee` or `waitForRedirectors`?

Reply:
Really, this all code is not going to work if any exception thrown.

The adding 

The process is not destroyed, and
the 'waitForRedirectors' is skipped in exception in waitForDebugee happens.
The 'waitForRedirectors' tries to close 3 redirectors and fail to close all of them if exception happens.
So all failure handling logic should be updated.
I changed it to try/finally so it close pipe, binder and destroy process.